src/pages/AddItem/index.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import Button from '../../components/button';
+import Title from '../../components/title';
+import ContentContainer from '../../components/content-container';
+import AddForm from '../../components/add-form';
+import { addProduct } from '../../utils/firebaseUtils';
+import './styles.css';
+
+export const AddItemPage = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [message, setMessage] = useState(null);
+  const [error, setError] = useState(false);
+  const [formState, setFormState] = useState({
+    productName: '',
+    timeFrame: '7',
+    lastPurchaseDate: null,
+  });
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    addProduct(formState)
+      .then((res) => console.log(res))
+      .catch(() => setError(true))
+      .finally(() => {
+        setIsLoading(false);
```

Comment:
The setIsLoading would always end up being false even if there's an error. This avoids you to declare that line in `.then` and `.catch`:+1::+1: :+1: